### PR TITLE
Advanced Trading View's Logic Update

### LINKED ISSUE #1581: Smooth animate new items in trade history
We do a fetch from different dexes' subgraph every 15 seconds.

It repaints the entire trade history. I think it will be good if we have a nicer way to do this so that only the new rows are added to the top. :)

https://images.zenhubusercontent.com/209400598/fa59eadc-3ad5-4854-ac4e-2aa20bd17539/screen_recording_2022_11_08_at_14_08_51.mov

### PR DESCRIPTION
**Pull request by bejzik8**
# Background

Resolves #1581.

The ticket above reports the issue of "repainting" the entire trade history. After thorough investigation, I came to the conclusion that there's more to it rather than just a UI update. The main issue is that the state was reseted every 15 seconds. Fetching logic needed to be rethought; I discovered that it could be optimised as the queries could be written better and get more data. Another issue was with **Redux** actions, the data was fetched in chunks as it is the only way considering it is coming from different endpoints but it could be dispatched to the reducer in one action and provide better optimisation and much less re-renderers; to be precise 10X less.

# Updates:

- added new queries for fetching data; now we get `swaps`, `burns` and `mints` together
- created new methods for data fetching
- added new types and updated existing ones
- optimised **Redux**, dispatch is now being handled better
- created new actions and removed the old ones
- updated `Advanced Trading View`'s adapter configuration
- created new methods in `Advanced Trading View`'s adapter
- updated `Advanced Trading View`'s hook which feeds the component with data
- removed unused types, methods and tests.

# In the Future

The most useful thing over here would be to add the documentation explaining the architecture, logic and all of the processes happening over here, both in front and in the background. Code could be split into smaller pieces by the functions it perform, logic could be separated from the UI and we could introduce some componentization.

🍀

### CODE REVIEW

File: src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts [hunk 1]
```diff
@@ -58,6 +65,71 @@ const advancedTradingViewSlice = createSlice({
...
+        const nextPairSwapData = state.adapters[key][pairId]?.[AdapterPayloadType.SWAPS]?.data ?? []
+        const nextPairBurnsAndMintsData = state.adapters[key][pairId]?.[AdapterPayloadType.BURNS_AND_MINTS]?.data ?? []
+
+        data.swaps.forEach(
+          element => !nextPairSwapData.some(el => el.id === element.id) && nextPairSwapData.push(element)
+        )
+        data.burnsAndMints.forEach(
+          element =>
+            !nextPairBurnsAndMintsData.some(el => el.id === element.id) && nextPairBurnsAndMintsData.push(element)
+        )
```
Comment (berteotti) [inline src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts:89]: Can we have a different name for `element` and `el`? Just to have a better reading in this code

File: src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts [hunk 2]
```diff
@@ -58,6 +65,71 @@ const advancedTradingViewSlice = createSlice({
...
+      const updatedAdapters: AdapterType = {
+        ...state.adapters,
+      }
+
+      action.payload.forEach(adapter => {
+        const { key, pairId, data, hasMore, payloadType } = adapter
+
+        const nextPairData = state.adapters[key][pairId]?.[payloadType]?.data ?? []
+
+        data.forEach(element => !nextPairData.some(el => el.id === element.id) && nextPairData.push(element))
```
Comment (berteotti) [inline src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts:120]: In this case too

**Review by berteotti — COMMENTED** [review]
Code looks good, just seems to have a TODO left and some ts-ignore that need to be addressed. Also left a few comments. Will see it again when PR is ready :)

File: src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts (hunk 1 above)
Comment (bejzik8) [reply src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts:89]: Agree, this is updated. 👍🏼

File: src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts (hunk 2 above)
Comment (bejzik8) [reply src/services/AdvancedTradingView/store/advancedTradingView.reducer.ts:120]: Agree. 👍🏼

**Comment by 0xVenky** [comment]
Just had a look, find that the list is not auto refreshing and fetching new trades 🤔  I know its only a draft, but couldnt wait :)

File: src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.test.ts [hunk 3]
```diff
@@ -74,193 +46,6 @@ describe('BaseAdapter', () => {
...
-        swaps: new Array(50).fill(FAKE_SWAP_DATA),
-      })
-    )
-
-    expect(graphqlRequestMock).toHaveBeenCalledTimes(0)
-
-    await baseAdapter.getPairTrades({
-      inputToken: USDC_TOKEN,
-      outputToken: USDT_TOKEN,
-      isFirstFetch: true,
```
Comment (Mi-Lan) [inline src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.test.ts:89]: should we adapt tests to new structure?

File: src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.ts [hunk 4]
```diff
@@ -102,22 +115,50 @@ export class BaseAdapter<
...
+          inputTokenAddress: inputToken.address,
+          outputTokenAddress: outputToken.address,
+        })
+
+        return {
+          payloadType: AdapterPayloadType.SWAPS,
+          key: this._key,
+          pairId,
+          data: swaps,
+          hasMore: swaps.length === amountToFetch,
```
Comment (Mi-Lan) [inline src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.ts:136]: does this tell us if it has more to fetch?

**Review by Mi-Lan — APPROVED** [review]
Code looks fine!

File: src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.test.ts (hunk 3 above)
Comment (bejzik8) [reply src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.test.ts:89]: I don't know. 🤷🏼

File: src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.ts (hunk 4 above)
Comment (bejzik8) [reply src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.ts:136]: Yeah, it is used when you scroll and look for the past data.

File: src/services/AdvancedTradingView/useAdvancedTradingView.hook.ts [hunk 5]
```diff
@@ -166,6 +155,13 @@ export const useAdvancedTradingView = () => {
     }
 
     fetchTrades()
+
+    const interval = setInterval(() => {
+      fetchTrades()
+    }, 15000)
```
Comment (berteotti) [inline src/services/AdvancedTradingView/useAdvancedTradingView.hook.ts:161]: I see we are using `15000` milliseconds all over our app. Don't know if it's better to have a global constant or just a constant for Advanced Trading View.

File: src/services/AdvancedTradingView/adapters/advancedTradingView.adapter.ts [hunk 6]
```diff
@@ -23,9 +26,9 @@ export abstract class AbstractAdvancedTradingViewAdapter<AppState> {
     this._adapterSupportedChains = adapterSupportedChains
   }
 
-  abstract getPairTrades(fetchDetails: AdapterFetchDetails): Promise<void>
+  abstract getPairTradingAndActivityData(fetchDetails: AdapterFetchDetails): any
 
-  abstract getPairActivity(fetchDetails: AdapterFetchDetails): Promise<void>
+  abstract getPairData(fetchDetails: AdapterFetchDetailsExtended): any
```
Comment (berteotti) [inline src/services/AdvancedTradingView/adapters/advancedTradingView.adapter.ts:31]: This return promises, right? Maybe we can use `Promise<void>` type here

**Review by berteotti — APPROVED** [review]
PR looks good, good job Mirko! Just left some small comments.

File: src/services/AdvancedTradingView/adapters/advancedTradingView.adapter.ts (hunk 6 above)
Comment (bejzik8) [reply src/services/AdvancedTradingView/adapters/advancedTradingView.adapter.ts]: This is resolved Leo.

File: src/services/AdvancedTradingView/useAdvancedTradingView.hook.ts (hunk 5 above)
Comment (bejzik8) [reply src/services/AdvancedTradingView/useAdvancedTradingView.hook.ts]: Done, created `REFETCH_DATA_INTERVAL` constant.

**Pull request merged**